dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -1310,105 +1307,57 @@ private StringBuilder dataElementAndFiltersSql(
           .append(" as ")
           .append(deUid);
 
-      String optValueTableAs = "opt_" + filterCount;
-
-      if (!joinedColumns.contains(deUid) && de.hasOptionSet() && !filters.isEmpty()) {
-        String optSetBind = "optset_" + filterCount;
-
-        mapSqlParameterSource.addValue(optSetBind, de.getOptionSet().getId());
-
-        optionValueJoinBuilder
-            .append("inner join optionvalue as ")
-            .append(optValueTableAs)
-            .append(" on lower(")
-            .append(optValueTableAs)
-            .append(".code) = ")
-            .append("lower(")
-            .append(dataValueValueSql)
-            .append(") and ")
-            .append(optValueTableAs)
-            .append(".optionsetid = ")
-            .append(":")
-            .append(optSetBind)
-            .append(" ");
-
-        joinedColumns.add(deUid);
-      }
-
-      if (!filters.isEmpty()) {
-        for (QueryFilter filter : filters) {
-          ++filterCount;
-
-          final String queryCol =
-              de.getValueType().isNumeric()
-                  ? castToNumber(dataValueValueSql)
-                  : lower(dataValueValueSql);
+      for (QueryFilter filter : filters) {
+        ++filterCount;
 
-          String bindParameter = "parameter_" + filterCount;
-          int itemType = de.getValueType().isNumeric() ? Types.NUMERIC : Types.VARCHAR;
+        final String queryCol =
+            de.getValueType().isNumeric()
+                ? castToNumber(dataValueValueSql)
+                : lower(dataValueValueSql);
 
-          if (!de.hasOptionSet()) {
-            eventDataValuesWhereSql.append(hlp.whereAnd());
+        String bindParameter = "parameter_" + filterCount;
+        int itemType = de.getValueType().isNumeric() ? Types.NUMERIC : Types.VARCHAR;
```

Comment:
This can be done above/outside the for loop right? Also suggesting itemValueType as the variable name?